src/util.rs

```diff
@@ -0,0 +1,14 @@
+// FIXME: Branch prediction hint. This is currently only available on nightly
```

Comment:
I wasn't feeling particularly creative and will need this in the `control` module too in the future, so, I decided to separate them into their own module.

Maybe `nightly` would be a better name, but I don't think it matters that much.

Reply:
It's actually broken at the moment on nightly until https://github.com/rust-lang/rust/pull/120370 is merged, so it doesn't even do anything. But it's probably fine to keep for now as a reminder to fix this later.